packages/loot-core/src/server/filters/app.ts

```diff
@@ -0,0 +1,182 @@
+      !filter.tombstone &&
+      filter.conditions.length === conditions.length
+    ) {
+      fCondCheck = false;
+      conditions.map((cond, i) => {
+        condCheck[i] =
+          filter.conditions.filter(fcond => {
+            return (
+              cond.value === fcond.value &&
+              cond.op === fcond.op &&
+              cond.field === fcond.field
+            );
+          }).length > 0;
+        fCondCheck = (i === 0 ? true : fCondCheck) && condCheck[i];
+        return true;
+      });
+      fCondFound = fCondCheck && condCheck[conditions.length - 1] && filter;
+    }
+    return true;
+  });
+
+  condCheck = [];
+
+  if (!newItem) {
+    return fCondFound
+      ? fCondFound.id !== item.id
+        ? fCondFound.name
+        : false
+      : false;
+  }
+  return fCondFound ? fCondFound.name : false;
+}
+
+async function createFilter(filter) {
+  let filterId = uuid.v4Sync();
+  let item = {
+    id: filterId,
+    conditions: filter.state.conditions,
+    conditionsOp: filter.state.conditionsOp,
+    name: filter.state.name,
+  };
+
+  if (item.name) {
+    if (await filterNameExists(item.name, item.id, true)) {
+      throw new Error('There is already a filter named ' + item.name);
+    }
+  } else {
+    throw new Error('Filter name is required');
+  }
+
+  if (item.conditions.length > 0) {
+    let condExists = conditionExists(item, filter.filters, true);
+    if (condExists) {
+      throw new Error(
+        'Duplicate filter warning: conditions already exist. Filter name: ' +
+          condExists,
+      );
+    }
+  } else {
+    throw new Error('Conditions are required');
+  }
+
+  // Create the filter here based on the info
+  await db.insertWithSchema('transaction_filters', filterModel.fromJS(item));
+
+  return filterId;
+}
+
+async function updateFilter(filter) {
+  let item = {
+    id: filter.state.id,
+    conditions: filter.state.conditions,
+    conditionsOp: filter.state.conditionsOp,
+    name: filter.state.name,
+  };
+  if (item.name) {
+    if (await filterNameExists(item.name, item.id, false)) {
+      throw new Error('There is already a filter named ' + item.name);
+    }
+  } else {
+    throw new Error('Filter name is required');
+  }
+
+  if (item.conditions.length > 0) {
+    let condExists = conditionExists(item, filter.filters, false);
+    if (condExists) {
+      throw new Error(
+        'Duplicate filter warning: conditions already exist. Filter name: ' +
+          condExists,
+      );
+    }
+  } else {
+    throw new Error('Conditions are required');
+  }
+
+  await db.updateWithSchema('transaction_filters', filterModel.fromJS(item));
+}
+
+async function deleteFilter({ id }) {
+  await batchMessages(async () => {
+    await db.delete_('transaction_filters', id);
+  });
+}
+
+app.method('filter/create', mutator(undoable(createFilter)));
+app.method('filter/update', mutator(undoable(updateFilter)));
+app.method('filter/delete', mutator(undoable(deleteFilter)));
```

Comment:
I'm not able to `ctrl+z` undo the deletion of a filter. Is this working as expected?

Reply:
The `/null` address is a known issue — @trevdor said he’s looking into fixing it: https://discord.com/channels/937901803608096828/1027831756545609789/1122964623121993880

Reply:
Thanks @trevdor. This does stop the /null from loading. However, I end up on a page with a value on my page that no longer exists in the back-end table. It gets loaded onto the front-end after the change is done then the undo reverts the change on the back-end but the value still shows as the new changed value on the front-end. Would this just require some sort of reload/refresh? Can we work that into the global-events element? Thanks!

Reply:
Would something like this cause errors for other windows/processes? This would fix my use case.
```
if (window.location.href !== tagged.url) {
            tagged.url ?
            window.location.href = tagged.url :
            window.location.reload();
            return true;
          }
```